examples/client/Locomotion/src/context/newRideContext/index.tsx

```diff
@@ -175,10 +188,18 @@ const RidePageContextProvider = ({ children }: {
     [RIDE_STATES.COMPLETED]: (completedRide: any) => {
       navigation.navigate(MAIN_ROUTES.POST_RIDE, { rideId: completedRide.id });
       changeBsPage(BS_PAGES.ADDRESS_SELECTOR);
-      cleanRideState();
+      cleanFullRideState();
     },
     [RIDE_STATES.DISPATCHED]: () => { changeBsPage(BS_PAGES.ACTIVE_RIDE); },
     [RIDE_STATES.ACTIVE]: () => { changeBsPage(BS_PAGES.ACTIVE_RIDE); },
+    [RIDE_STATES.CANCELED]: (canceledRide: any) => {
+      if (canceledRide.canceledBy !== user?.id) {
+        setRidePopup(RIDE_POPUPS.RIDE_CANCELED_BY_DISPATCHER);
```

Comment:
maybe also set the stop points to the request stop points? 🤔

Reply:
i set everything needed on the popup buttons